tips/tip-0001/tip-0001.md

```diff
@@ -0,0 +1,67 @@
+
+## Contiguity Argument
+
+The contiguity argument is only needed for the Ram Table because the memory pointer there (`ramp`) can take any value. In contrast, the two other memory-like tables, OpStackTable and JumpStackTable are stacks and therefore naturally generate contiguous access patterns.
+
+Since the contiguity argument may have applications elsewhere, it is presented here in a generic language.
+
+Let `ramp` be the column whose contiguity we wish to establish. We add three extension columns: the difference inverse `di`, the running product `rp`, and the formal derivative `fd`. Additionally, the prover commits to two polynomials $a(X)$ and $b(X)$ but these polynomials do not correspond to columns in the table.
+
+The values contained in these columns is undetermined until the verifier's challenge $\alpha$ is known; before that happens it is worthwhile to present the polynomial expressions in $X$, anticipating the substitution $X \mapsto \alpha$.
+
+The difference inverse takes the inverse of the difference between the current and next `ramp` values. This constraint corresponds to two consistency constraint polynomials: $(\mathsf{ramp}^\star - \mathsf{ramp})\cdot((\mathsf{ramp}^\star - \mathsf{ramp}) \cdot \mathsf{di} - 1)$ and $\mathsf{di}\cdot((\mathsf{ramp}^\star - \mathsf{ramp}) \cdot \mathsf{di} - 1)$.
+
+Let $g$ and $h$ be uniformly random scalars. These scalars will be used as initials to randomize the `rp` and `fd` columns when zero-knowledge is an objective. When zero-knowledge is not necessary, it cannot hurt to set $g = 1$ and $h=0$. Also note that since $h$ does not need to be the formal derivative of $g$, and if this is the case the name "formal derivative" for the column is not entirely accurate. However, it may help at first reading to imagine these scalars as polynomials satisfying $h(X) = g'(X)$.
+
+The running product starts with $g$ initially and accumulates a factor $X - \mathsf{ramp}$ in every pair of rows where $\mathsf{ramp} \neq \mathsf{ramp}^\star$. This evolution corresponds to one transition constraint: $(\mathsf{ramp}^\star - \mathsf{ramp}) \cdot (\mathsf{rp}^\star - \mathsf{rp} \cdot (X - \mathsf{ramp})) + (1 -(\mathsf{ramp}^\star -\mathsf{ramp}) \cdot \mathsf{di}) \cdot (\mathsf{rp}^\star - \mathsf{rp})$.
+
+The associated terminal value is $T_{\mathsf{rp}}$ and the terminal constraint is stipulates that the factor $X - \mathsf{ramp}$ from the last row is accumulated as well: $\mathsf{rp} \cdot (X - \mathsf{ramp}) - T_{\mathsf{rp}}$. Alternatively, if the last row is a padding row and should not be accumulated into the running product, then the last value in the column is already the terminal value: $\mathsf{rp} - T_{\mathsf{rp}}$.
```

Comment:
In Triton, the memory tables can easily be derived after padding has happened, implying that any permutation arguments between the processor and the respective memory table will work out. Are you presenting padding rows here in order to make the contiguity argument more generic, i.e., easier to apply beyond Triton VM?